apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/utility/DataUtilityCommons.java

```diff
@@ -154,14 +156,17 @@ public static void unzip(Path zipFilePath, Path destDirectory) throws IOExceptio
    * @throws IOException (any errors encountered will be bubbled up)
    */
   public static void extractFile(ZipInputStream zipIn, Path filePath) throws IOException {
-    BufferedOutputStream bos =
-        new BufferedOutputStream(new FileOutputStream(filePath.toFile().getAbsolutePath()));
-    byte[] bytesIn = new byte[BUFFER_SIZE];
-    int read = 0;
-    while ((read = zipIn.read(bytesIn)) != -1) {
-      bos.write(bytesIn, 0, read);
+    Files.createDirectories(filePath.getParent());
+    try (BufferedOutputStream bos =
+        new BufferedOutputStream(new FileOutputStream(filePath.toFile().getAbsolutePath()))) {
+      byte[] bytesIn = new byte[BUFFER_SIZE];
+      int read;
+      while ((read = zipIn.read(bytesIn)) != -1) {
+        bos.write(bytesIn, 0, read);
+      }
     }
-    bos.close();
+
+    zipIn.closeEntry();
```

Comment:
Do you need to do this here if it's already being done in the calling method? Maybe this method ought to take `final ZipEntry entry` instead of the `ZipInputStream`

Reply:
Removed that call as it isn't necessary. The close method of the BufferedOutputStream object is also redudant because of the try resource.